### PR TITLE
[API] Resend verification email refactor to take current user from the security context

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php [hunk 1]
```diff
@@ -16,30 +16,24 @@
 use Sylius\Bundle\ApiBundle\Command\ChannelCodeAwareInterface;
 use Sylius\Bundle\ApiBundle\Command\IriToIdentifierConversionAwareInterface;
 use Sylius\Bundle\ApiBundle\Command\LocaleCodeAwareInterface;
+use Sylius\Bundle\ApiBundle\Command\ShopUserIdAwareInterface;
 
 /** @experimental */
-class ResendVerificationEmail implements ChannelCodeAwareInterface, LocaleCodeAwareInterface, IriToIdentifierConversionAwareInterface
+class ResendVerificationEmail implements ShopUserIdAwareInterface, ChannelCodeAwareInterface, LocaleCodeAwareInterface, IriToIdentifierConversionAwareInterface
```
Comment (stloyd) [inline src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php:22]:
```suggestion
final class ResendVerificationEmail implements ShopUserIdAwareInterface, ChannelCodeAwareInterface, LocaleCodeAwareInterface, IriToIdentifierConversionAwareInterface
```

File: src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php [hunk 2]
```diff
@@ -16,30 +16,24 @@
...
-    public $email;
+    /** @var mixed */
+    public $shopUserId;
 
     /**
-     * @var string
-     * @psalm-immutable
+     * @var string|null
      */
     public $channelCode;
```
Comment (stloyd) [inline src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php:30]: It's PHP 7.4 baby 😂 
```suggestion
    public ?string $channelCode = null;
```

File: src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php [hunk 3]
```diff
@@ -16,30 +16,24 @@
...
 use Sylius\Bundle\ApiBundle\Command\LocaleCodeAwareInterface;
+use Sylius\Bundle\ApiBundle\Command\ShopUserIdAwareInterface;
 
 /** @experimental */
-class ResendVerificationEmail implements ChannelCodeAwareInterface, LocaleCodeAwareInterface, IriToIdentifierConversionAwareInterface
+class ResendVerificationEmail implements ShopUserIdAwareInterface, ChannelCodeAwareInterface, LocaleCodeAwareInterface, IriToIdentifierConversionAwareInterface
 {
-    /** @var string */
-    public $email;
+    /** @var mixed */
```
Comment (stloyd) [inline src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php:24]: Shouldn't be:
```suggestion
    /** @var int|string|null */
```

File: src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php (hunk 1 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php:22]: I agree, that this approach would be more proper from the clean code perspective, but we've decided to leave commands open, as they are simple DTOs and it makes it easier to extend basic logic

File: src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php (hunk 2 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php:30]: Tempting, but I'm not sure, we should do it before the main effort. /cc @Zales0123 @Arminek @GSadee

File: src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php (hunk 3 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/ApiBundle/Command/Account/ResendVerificationEmail.php:24]: could be

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/VerifyCustomerAccount.xml [hunk 4]
```diff
@@ -33,6 +33,9 @@
                 <attribute name="openapi_context">
                     <attribute name="summary">Resends verification email</attribute>
                 </attribute>
+                <attribute name="security">
+                    is_granted("ROLE_USER")
+                </attribute>
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/VerifyCustomerAccount.xml:38]: Can be in one line

File: src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Taxon.xml [hunk 5]
```diff
@@ -19,5 +19,8 @@
             <group>admin:taxon:read</group>
             <group>shop:taxon:read</group>
         </attribute>
+        <attribute name="children">
+            <group>shop:taxon:read</group>
+        </attribute>
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Taxon.xml:24]: Is it needed for this PR?

**Review by GSadee — APPROVED** [review]
But the rebase is needed

**Comment by GSadee** [comment]
Thanks, Łukasz! :tada:

**Pull request merged**